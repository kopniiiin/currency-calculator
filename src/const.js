export const Currency = {
  EUR: `EUR`,
  USD: `USD`,
  JPY: `JPY`
};

export const currenciesToTranslations = {
  [Currency.EUR]: `ÐµÐ²Ñ€Ð¾`,
  [Currency.USD]: `Ð´Ð¾Ð»Ð»Ð°Ñ€`,
  [Currency.JPY]: `Ð¸ÐµÐ½Ð°`
};

export const currenciesToEmoji = {
  [Currency.EUR]: `ðŸ’¶`,
  [Currency.USD]: `ðŸ’µ`,
  [Currency.JPY]: `ðŸ’´`
};

export const RATE_PRECISION = 2;
