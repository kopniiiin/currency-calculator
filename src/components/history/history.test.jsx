import React from "react";
import {shallow} from "enzyme";

import {Currency} from "../../const.js";

import History from "./history.jsx";

describe(`snapshot test: History`, () => {
  it(`should render correctly`, () => expect(
    shallow(
      <History
        operations={[{
          id: `4`,
          date: new Date(`2020-04-04`),
          originalCurrency: Currency.EUR,
          finalCurrency: Currency.EUR,
          originalCurrencyAmount: 4,
          finalCurrencyAmount: 4
        }]}/>
    )
  ).toMatchSnapshot());
});
