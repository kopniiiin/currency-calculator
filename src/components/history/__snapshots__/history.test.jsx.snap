// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: History should render correctly 1`] = `
<ol
  className=" history"
>
  <li
    className="history__item"
    key="4"
  >
    <time
      className="history__date"
      dateTime="2020-04-04"
    >
      2020-04-04
    </time>
    <br />
    <span
      aria-label="ÐµÐ²Ñ€Ð¾"
      className="history__currency"
    >
      ðŸ’¶
    </span>
    Â 
    4.00
     =&gt; 
    <span
      aria-label="ÐµÐ²Ñ€Ð¾"
      className="history__currency"
    >
      ðŸ’¶
    </span>
    Â 
    4.00
  </li>
</ol>
`;
